yoti-sdk/src/main/java/com/yoti/mobile/android/sdk/YotiSDKButton.java

```diff
@@ -135,6 +168,10 @@ public void onClick(View v) {
         void onYotiAppNotInstalledError(YotiSDKNoYotiAppException cause);
     }
 
+    public interface OnAppNotInstalledListener {
+        void onAppNotInstalled(YotiSDKAppNotInstalledException cause, String appURL);
+    }
+
     public interface OnYotiCalledListener {
```

Comment:
🤔  this is only called when Yoti App is launched? So, what if we're calling Easy ID or other one?

Reply:
Yeah, based on the applied theme we need to launch the specific app. We will be notifying integrator with this listener whenever user taps the button.

Reply:
Then I think we also need to deprecate it and create a new one called `OnAppCalledListener`, now seems like it only communicates `Yoti` app states and also, just for the sake of consistency :)

Reply:
Deprecated the `setOnYotiCalledListener` and added `setOnAppCalledListener`. And related to passing the app type when app is being called, for EasyID theme we can pass the app type, but when integrator chooses Yoti or partnership theme, user can choose any app through launcher, so difficult to determine which app is being selected by the user.

Reply:
🎉  I'm checking now sample apps Main Activities and there are some deprecated calls to `setOnYotiAppNotInstalledListener` and `setOnYotiScenarioListener`. I think we should also correct them :)